contracts/bots/PartialLiquidationBot.sol

```diff
@@ -0,0 +1,280 @@
+        CollateralDebtData memory cdd = ICreditManagerV3(params.creditManager).calcDebtAndCollateral(
+            params.creditAccount, CollateralCalcTask.DEBT_COLLATERAL
+        );
+
+        if (cdd.twvUSD >= cdd.totalDebtUSD) revert CreditAccountNotLiquidatableException();
+
+        params.totalDebt = CreditLogic.calcTotalDebt(cdd);
+    }
+
+    /// @dev Internal function implementing the main liquidation logic
+    ///      1. Determines the maximal amount of underlying that can be sold and maximal amount of assetOut that can be bought
+    ///      2. Computes the input and output amounts
+    ///      3. Transfers underlying from caller to Credit Account
+    ///      4. Performs a multicall to enable the underlying and transfer assetOut to caller (and optionally decrease debt)
+    ///      5. Checks that the resulting HF is above the minimal threshold
+    function _liquidate(LiquidationParams memory params) internal {
+        uint256 maxAmountIn;
+        uint256 maxAmountOut;
+        {
+            uint256 underlyingBalance = IERC20(params.underlying).balanceOf(params.creditAccount);
+            (uint256 minDebt,) = ICreditFacadeV3(params.creditFacade).debtLimits();
+
+            uint256 repayable;
+
+            // If the liquidation is with repayment, the maximal amount of underlying is exactly enough to repay until minDebt
+            // If there is no repayment, we allow to fill up the account up to the totalDebt + a 1% buffer to cover future interest
+
+            if (params.repay) {
+                repayable = params.totalDebt - minDebt;
+            } else {
+                repayable = params.totalDebt * DEBT_BUFFER / PERCENTAGE_FACTOR;
+            }
+
+            if (underlyingBalance >= repayable) revert NothingToLiquidateException();
+
+            maxAmountIn = repayable - underlyingBalance;
+            maxAmountOut = IERC20(params.assetOut).balanceOf(params.creditAccount);
+        }
+
+        if (params.exactIn) {
+            (params.amountIn, params.amountOut) = _getAmountsExactIn(params, maxAmountIn, maxAmountOut);
+        } else {
+            params.amountOut = params.amountOut == type(uint256).max
+                ? IERC20(params.assetOut).balanceOf(params.creditAccount)
+                : params.amountOut;
+            (params.amountIn, params.amountOut) = _getAmountsExactOut(params, maxAmountIn, maxAmountOut);
+        }
+
+        IERC20(params.underlying).transferFrom(msg.sender, params.creditAccount, params.amountIn);
+
+        ICreditFacadeV3(params.creditFacade).botMulticall(params.creditAccount, _getMultiCall(params));
+
+        CollateralDebtData memory cdd = ICreditManagerV3(params.creditManager).calcDebtAndCollateral(
+            params.creditAccount, CollateralCalcTask.DEBT_COLLATERAL
+        );
+
+        if (cdd.twvUSD * PERCENTAGE_FACTOR < cdd.totalDebtUSD * THRESHOLD_HEALTH_FACTOR) {
+            revert HealthFactorTooLowException();
```

Comment:
Maybe we can pass `minHealthFactor` to full collateral check and then repeat the calculation if `creditManager.version() == 3_00`? would save considerable amounts of gas for all new credit managers

Reply:
Fixed